tests/ci-cluster/gke.ts

```diff
@@ -26,11 +26,11 @@ export class GkeCluster extends pulumi.ComponentResource {
                 opts: pulumi.ComponentResourceOptions = {}) {
         super("pulumi-kubernetes:ci:GkeCluster", name, {}, opts);
 
-        // Find the latest 1.19.x engine version.
+        // Find the latest 1.23.x engine version.
         const engineVersion = gcp.container.getEngineVersions({
             location: config.gcpLocation,
             project: config.gcpProject,
-        }).then(v => v.validMasterVersions.filter(v => v.startsWith("1.19"))[0]);
+        }).then(v => v.validMasterVersions.filter(v => v.startsWith("1.23"))[0]);
```

Comment:
Adding a note to link to https://github.com/pulumi/pulumi-kubernetes/issues/1862 so we can add such tests to a matrix. Hopefully we can also drop down the number of tests that require a cloud K8s cluster to make such a matrix more feasible. We should be able to run all tests with a min and max version of kubernetes to support simultaneously with kind + GKE clusters.